### DISCUSSION TITLE
General Question about learning

### DISCUSSION THREAD

**Discussion by xahmd** [discordjs/discord.js#8056] [General]
From ur opinion what's the best way to learn discordjs library for beginners 
Please share ur ideas , iam going to make a research on the answers 

![a_b7a4d67b969d52377f12af9ff0afdccc](https://user-images.githubusercontent.com/79005095/172926736-0db1758c-f3da-468e-a2d0-96513ba92bad.gif)

**Comment by almeidx** [comment]
https://discordjs.guide/

**Comment by poggy-fish** [comment]
Open Source codes and the doc

**Reply by xahmd** [reply]
Thank you, that's a great answer !

**Comment by sebastianjnuwu** [comment]
In my opinion the best ways:
Learn the basics: https://discordjs.guide
Possible errors encountered by beginners here is the solution: https://stackoverflow.com
Learn the Intermediate: https://discord.js.org/#/docs/discord.js
For inquiries or questions: https://github.com/discordjs/discord.js/discussions

**Comment by padlocks** [comment]
Aside from what has already been suggested, learning by example is also a great way to do it. I often search for repositories on GitHub just to see what new stuff I can find and utilize. Discord.js has a lot of documentation, open source examples, and Stack Overflow content, so if you know what you're looking for, it shouldn't be too difficult to find.

**Comment by abdeljalil-salhi** [comment]
You can check https://discordjs.guide/#before-you-begin to begin with,
and watch a youtube video to understand the logics, then you can do your own tests to get what you want !
example [create a music bot using discordjs](https://youtu.be/fN29HIaoHLU)

**Comment by Gamers-geek** [comment]
In my opinion, I think [discordjs.guide](https://discordjs.guide) is a good website to learn discordjs
To improve your knowledge, you can also check [the docs](https://discord.js.org) and the [@reconlx](https://github.com/reconlx)'s [YouTube Channel](https://www.youtube.com/c/reconlxx) which is a good channel to learn discordjs
